File: CS50/Week 2/readability.c
#include <ctype.h>
#include <cs50.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    // string text = get_string("Text: ");
    string text = ("Hello, World!");

    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);

    // calculate L and S
    
   float L = ((float)letters / words) * 100;
   float S = ((float)sentences / words) * 100;
   // calculate indes
   float index = 0.0588 * L - 0.296 * S - 15.8;
   int grade = round(index);


    // print final grade
    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }
}


int count_letters(string text)
{
    int count =  0;
    for (int i = 0, length = strlen(text); i < length; i++)
    {
        if (isalpha(text[i]))
        {
            count ++;
        }
    }
    return count;
}

int count_words(string text)
{
    int count =  1;
    for (int i = 0, length = strlen(text); i < length; i++)
    {
        if (text[i] == ' ')
        {
            count ++;
        }
    }
    return count;
}

int count_sentences(string text)
{
    int count =  0;
    for (int i = 0, length = strlen(text); i < length; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            count ++;
        }
    }
    return count;
}

File: CS50/Week 6/readability.py
import re
text = input('Text: ')

word_no = len(text.split(' '))
sentence_no = len(re.split(r'[.!?]',text)) -1
char_no = 0
for i in text:
    if i.isalpha() or i.isdigit():
        char_no += 1

L = (char_no / word_no) * 100
S = (sentence_no / word_no) * 100
index = round(0.0588 * L - 0.296 * S - 15.8)


if index < 1:
    print("Before Grade 1")
elif index >= 16:
    print("Grade 16+")
else:
    print(f'Grade {index}')
